transports/webrtc/src/tokio/connection.rs

```diff
@@ -0,0 +1,312 @@
+                                                    );
+                                                }
+                                            }
+                                        }
+                                        Err(e) => {
+                                            log::error!("Can't detach data channel: {}", e);
+                                        }
+                                    };
+                                })
+                            })
+                        })
+                        .await;
+                })
+            }))
+            .await;
+    }
+}
+
+impl StreamMuxer for Connection {
+    type Substream = Substream;
+    type Error = Error;
+
+    fn poll_inbound(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Self::Substream, Self::Error>> {
+        match ready!(self.incoming_data_channels_rx.poll_next_unpin(cx)) {
+            Some(detached) => {
+                log::trace!("Incoming substream {}", detached.stream_identifier());
+
+                let (substream, drop_listener) = Substream::new(detached);
+                self.drop_listeners.push(drop_listener);
+                if let Some(waker) = self.no_drop_listeners_waker.take() {
+                    waker.wake()
+                }
+
+                Poll::Ready(Ok(substream))
+            }
+            None => {
+                debug_assert!(
+                    false,
+                    "Sender-end of channel should be owned by `RTCPeerConnection`"
+                );
+
+                Poll::Pending // Return `Pending` without registering a waker: If the channel is closed, we don't need to be called anymore.
+            }
+        }
+    }
+
+    fn poll(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<StreamMuxerEvent, Self::Error>> {
+        loop {
+            match ready!(self.drop_listeners.poll_next_unpin(cx)) {
+                Some(Ok(())) => {}
+                Some(Err(e)) => {
+                    log::debug!("a DropListener failed: {e}")
+                }
+                None => {
+                    self.no_drop_listeners_waker = Some(cx.waker().clone());
+                    return Poll::Pending;
+                }
+            }
+        }
+    }
+
+    fn poll_outbound(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Self::Substream, Self::Error>> {
+        let peer_conn = self.peer_conn.clone();
+        let fut = self.outbound_fut.get_or_insert(Box::pin(async move {
+            let peer_conn = peer_conn.lock().await;
+
+            // Create a datachannel with label 'data'
+            let data_channel = peer_conn.create_data_channel("", None).await?;
+
+            log::trace!("Opening outbound substream {}", data_channel.id());
+
+            // No need to hold the lock during the DTLS handshake.
+            drop(peer_conn);
```

Comment:
```suggestion
            // No need to hold the lock during the DTLS handshake.
            drop(peer_conn);
            
            log::trace!("Opening outbound substream {}", data_channel.id());
```

Nit pick to drop the lock guard as early as possible.